[scroll-animations] WPT tests for `timeline-scope` parsing are failures

graouts opened the pull request:
#### 624fe4c20d6c21d9223f5c93a93f1413cd6244c7
<pre>
[scroll-animations] WPT tests for `timeline-scope` parsing are failures
<a href="https://bugs.webkit.org/show_bug.cgi?id=287415">https://bugs.webkit.org/show_bug.cgi?id=287415</a>

Reviewed by Tim Nguyen.

The tests for the `timeline-scope` property were outdated and needed an update
to match the current `timeline-scope` grammar which uses `&lt;dashed-ident&gt;`. Fixing
those tests revealed an issue with `TimelineScope::operator==` which made computed
style not appear dirty and make `scroll-animations/css/timeline-scope-computed.tentative.html`
fail. Now both tests are up-to-date and pass.

* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/timeline-scope-computed.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/timeline-scope-computed.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/timeline-scope-parsing.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/timeline-scope-parsing.tentative.html:
* Source/WebCore/animation/TimelineScope.h:
(WebCore::TimelineScope::operator== const):

Canonical link: <a href="https://commits.webkit.org/290186@main">https://commits.webkit.org/290186@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/66b883c21f04600065971d9970786cbdf530d0f0

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/89170 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/8694 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/43997 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/94162 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/39931 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/91221 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/9082 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/16883 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/68717 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/26383 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/92172 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/6965 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/80935 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/49079 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/6714 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/35313 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/39038 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/77077 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/36300 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/95984 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/16350 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/11990 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [⏳ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/GTK-WK2-Tests-EWS "Waiting to run tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/16606 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/76721 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/76882 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/21288 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/19897 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/9465 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/13983 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/16364 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/21675 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/16105 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/19556 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/17886 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/animation/TimelineScope.h:
```diff
@@ -34,7 +34,7 @@ struct TimelineScope {
-            return other.type == Type::Ident && scopeNames != other.scopeNames;
+            return other.type == Type::Ident && scopeNames == other.scopeNames;
         return type == other.type;
     }
```
Just using the default operator would probably have avoided this bug.

graouts replied:
Indeed, that works as well, I'll use the default implementation.

graouts committed:
[scroll-animations] WPT tests for `timeline-scope` parsing are failures
https://bugs.webkit.org/show_bug.cgi?id=287415

Reviewed by Tim Nguyen.

The tests for the `timeline-scope` property were outdated and needed an update
to match the current `timeline-scope` grammar which uses `<dashed-ident>`. Fixing
those tests revealed an issue with `TimelineScope::operator==` which made computed
style not appear dirty and make `scroll-animations/css/timeline-scope-computed.tentative.html`
fail. Now both tests are up-to-date and pass.

* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/timeline-scope-computed.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/timeline-scope-computed.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/timeline-scope-parsing.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/timeline-scope-parsing.tentative.html:
* Source/WebCore/animation/TimelineScope.h:
(WebCore::TimelineScope::operator== const):

Canonical link: https://commits.webkit.org/290186@main
